### PR TITLE
fix: filtering toolbar disappearing

### PR DESCRIPTION
**Pull request by nategrift**
## Problem
Issue: #16638 

When all feature flags are enabled and you filter down the list to show only ones that are disabled, the filter toolbar disappears until you refresh the page. This bug also occurs if you have all your feature flags as disabled and filter down to only enabled flags.

## Changes

- Only show "Create Feature Flag" popup if there are no flags (without filtering)
- Display a different empty state when flags exist but are filtered out to inform user that they need to change their filters.

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?
- Local Testing
   - Checked filtering doesnt remove toolbar
   - Confirm that original "Create Feature Flag" empty state shows when no popup exists

### CODE REVIEW

File: frontend/src/scenes/feature-flags/featureFlagsLogic.ts [hunk 1]
```diff
@@ -132,9 +132,15 @@ export const featureFlagsLogic = kea<featureFlagsLogicType>({
...
-            (featureFlagsLoading, searchedFeatureFlags, searchTerm): boolean => {
-                return searchedFeatureFlags && searchedFeatureFlags?.length == 0 && !featureFlagsLoading && !searchTerm
+            (s) => [s.featureFlagsLoading, s.searchedFeatureFlags, s.searchTerm, s.filters],
+            (featureFlagsLoading, searchedFeatureFlags, searchTerm, filters): boolean => {
+                return (
+                    searchedFeatureFlags &&
+                    searchedFeatureFlags?.length == 0 &&
+                    !featureFlagsLoading &&
+                    !searchTerm &&
+                    !filters.active
```
Comment (neilkakkar) [inline frontend/src/scenes/feature-flags/featureFlagsLogic.ts:142]: Hey @nategrift , this same bug exists not just for the active filter, but any other kind of filter too. So, worth solving in this same PR.

(i.e. if you choose created by = person who didn't create any flags, same problem)

Comment (nategrift) [reply frontend/src/scenes/feature-flags/featureFlagsLogic.ts:142]: Okay! Will do

**Review by neilkakkar — APPROVED** [review]
much better!

**Comment by nategrift** [comment]
@posthog-bot Just waiting on review, not a stale pr.

**Review by liyiy — APPROVED** [review]

**Pull request merged**